s3-handler/main.go

```diff
@@ -106,29 +106,8 @@ func Handler(request events.S3Event) (Response, error) {
 
 			// handle XRay formatted trace id
 			if tmp, ok := parsedLine["trace_id"]; ok {
-				tmpString := tmp.(string)
-				if strings.Contains(tmpString, "Root=") {
-					parts := strings.Split(tmpString, ";")
-					for _, v := range parts {
-						p := strings.Split(v, "=")
-						if len(p) > 1 {
-							key := p[0]
-							switch key {
-							case "Root":
-								parsedLine["trace_id"] = p[1]
-							case "Self":
-								parsedLine["trace_id.self"] = p[1]
-							case "Parent":
-								parsedLine["trace_id.parent"] = p[1]
-							case "Sampled":
-								parsedLine["trace_id.sampling_decision"] = p[1]
-							default:
-								// this accounts for custom fields that may end up in the X-Amzn-Trace-Id field
-								parsedLine["headers."+key] = p[1]
-							}
-						}
-					}
-				}
+				ids := parseXRayTraceID(tmp.(string))
+				maps.Copy(parsedLine, ids)
```

Comment:
Pass in `parsedLine` here so you can add directly to it, saving the alloc and copy?